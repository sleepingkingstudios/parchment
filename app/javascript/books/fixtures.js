import { load } from '../fixtures';

/* eslint-disable-next-line import/prefer-default-export */
export const booksData = load('books');
