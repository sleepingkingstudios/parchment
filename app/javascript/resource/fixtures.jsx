import React from 'react';

export const Block = () => (<div />);
export const Form = () => (<div />);
export const Table = () => (<div />);

export const CreatePage = () => (<div />);
export const IndexPage = () => (<div />);
export const LandingPage = () => (<div />);
export const ShowPage = () => (<div />);
export const UpdatePage = () => (<div />);

export const fixtures = {
  create: {
    component: CreatePage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/create',
    reducer: ((state = { widget: {} }) => state),
  },
  index: {
    component: IndexPage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/',
    reducer: ((state = { widgets: [] }) => state),
  },
  show: {
    component: ShowPage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/:id',
    reducer: ((state = { widget: {} }) => state),
  },
  update: {
    component: UpdatePage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/:id/update',
    reducer: ((state = { widget: {} }) => state),
  },
};
