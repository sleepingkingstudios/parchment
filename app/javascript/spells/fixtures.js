/* eslint-disable-next-line import/prefer-default-export */
export const spellsData = [
  {
    id: 'e17bca01-24d3-48d3-a308-099dd6dca373',
    level: 0,
    name: 'Embers',
    school: 'Evocation',
  },
  {
    id: 'e3641970-fc98-4674-b873-5532d89d9f26',
    level: 3,
    name: "Dragon's Breath",
    school: 'Evocation',
  },
  {
    id: 'a74d8c21-607d-4a3f-8cfd-2bbb48ec196a',
    level: 9,
    name: 'Wrath of the Titans',
    school: 'Evocation',
  },
];
