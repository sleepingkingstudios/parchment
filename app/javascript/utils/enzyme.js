import { valueOrDefault } from './object';
import { safeCapitalize } from './string';

export const extractInjectedComponent = (component) => {
  const injected = component.shallow().find('ComponentWithInjectedProps');

  return injected.exists() ? injected : component;
};

export const toHaveFormControl = expect => (received, expectedControl, withProps) => {
  const formControl = received.find(expectedControl);
  const expectedProps = valueOrDefault(withProps, {});

  expect(formControl).toExist();
  expect(formControl).toHaveProp(expectedProps);

  return { pass: true, message: () => '' };
};

export const toHaveFormGroup = expect => (received, expectedGroup, fn) => {
  const formGroup = received.find(expectedGroup);

  expect(formGroup).toExist();

  fn(extractInjectedComponent(formGroup).shallow());

  return { pass: true, message: () => '' };
};

export const toHaveFormInput = expect => (received, props) => {
  const {
    form,
    inputType,
    prop,
  } = props;

  expect(received).toHaveProp({ prop });

  const formInput = received.find(`${safeCapitalize(prop)}Input`);

  expect(formInput).toExist();
  expect(formInput).toHaveProp({ form });

  const shallowInput = formInput.shallow();

  expect(shallowInput).toExist();
  expect(shallowInput).toHaveDisplayName(valueOrDefault(inputType, 'FormInput'));

  return { pass: true, message: () => '' };
};
