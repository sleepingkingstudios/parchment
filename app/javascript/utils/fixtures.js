/* eslint-disable-next-line import/prefer-default-export */
export const trimMultilineLiteral = (text) => {
  if (!text) { return ''; }

  const paragraphs = text.split('\n\n');

  return paragraphs
    .map((paragraph) => {
      const lines = paragraph.split('\n');

      return lines.map(line => line.trim()).join(' ');
    })
    .join('\n\n');
};
