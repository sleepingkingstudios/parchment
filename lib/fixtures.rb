# frozen_string_literal: true

# Namespace for fixtures, which handle reading data from a serialized source and
# hydrating the database.
module Fixtures
  def self.build(record_class, count: nil, environment: 'fixtures')
    Builder.new(record_class, environment: environment).build(count: count)
  end

  def self.create(record_class, count: nil, environment: 'fixtures')
    Builder.new(record_class, environment: environment).create(count: count)
  end

  def self.read(record_class, count: nil, environment: 'fixtures')
    Builder.new(record_class, environment: environment).read(count: count)
  end

  autoload :Builder, 'fixtures/builder'
end
