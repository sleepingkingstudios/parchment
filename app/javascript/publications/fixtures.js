import { load } from '../fixtures';

const isOfficial = (publication) => {
  const { playtest, publisherName } = publication;

  return !playtest && publisherName === 'Wizards of the Coast';
};

const rawData = load('publications');

/* eslint-disable-next-line import/prefer-default-export */
export const publicationsData = rawData
  .map(publication => Object.assign(
    {},
    publication,
    {
      official: isOfficial(publication),
    },
  ));
