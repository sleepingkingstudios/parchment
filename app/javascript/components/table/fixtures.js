import { humanizeList } from '../../utils/array';

export const formatCrewCapacity = (item) => {
  const count = item.crewCapacity;

  if (count === 1) { return '1 Kerbal'; }

  return `${count} Kerbals`;
};

export const formatExperiments = (item) => {
  humanizeList(item.experiments, { default: 'None' });
};

export const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Crew Capacity',
    prop: 'crewCapacity',
    value: formatCrewCapacity,
  },
  {
    label: 'Experiments',
    prop: 'experiments',
  },
];

export const rocketsData = [
  {
    id: '83f5e9e8-92df-4203-bc2e-4381370c2186',
    name: 'Apprentice',
    crewCapacity: 1,
    experiments: [
      'Crew Report',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '4bd7454c-3b15-4047-8d57-dfa831dac58e',
    name: 'Sorcerer',
    crewCapacity: 2,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '2fead083-d9ea-4f04-949e-d8c00e800cbb',
    name: 'Warlock',
    crewCapacity: 3,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
      'Gravity Scan',
    ],
  },
];
