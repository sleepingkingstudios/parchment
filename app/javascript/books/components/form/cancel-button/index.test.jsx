import React from 'react';
import { shallow } from 'enzyme';

import BookFormCancelButton from './index';

import { booksData } from '../../../fixtures';

describe('<BookFormCancelButton />', () => {
  const book = booksData[0];
  const form = { data: { book } };
  const defaultProps = {
    form,
    isUpdate: false,
  };

  it('should render the button', () => {
    const rendered = shallow(<BookFormCancelButton {...defaultProps} />);
    const url = '/books';

    expect(rendered).toHaveDisplayName('LinkButton');
    expect(rendered).toHaveProp('block', true);
    expect(rendered).toHaveProp('outline', true);
    expect(rendered).toHaveProp({ url });
    expect(rendered).toHaveProp('children', 'Cancel');
  });

  it('should match the snapshot', () => {
    const rendered = shallow(<BookFormCancelButton {...defaultProps} />);

    expect(rendered).toMatchSnapshot();
  });

  describe('with isUpdate: true', () => {
    const props = { ...defaultProps, isUpdate: true };

    it('should render the button', () => {
      const rendered = shallow(<BookFormCancelButton {...props} />);
      const url = `/books/${book.id}`;

      expect(rendered).toHaveDisplayName('LinkButton');
      expect(rendered).toHaveProp('block', true);
      expect(rendered).toHaveProp('outline', true);
      expect(rendered).toHaveProp({ url });
      expect(rendered).toHaveProp('children', 'Cancel');
    });
  });
});
