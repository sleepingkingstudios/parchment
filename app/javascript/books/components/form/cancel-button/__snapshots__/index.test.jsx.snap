// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookFormCancelButton /> should match the snapshot 1`] = `
<LinkButton
  block={true}
  buttonSize={null}
  buttonStyle="secondary"
  className={null}
  link={false}
  outline={true}
  url="/books"
>
  Cancel
</LinkButton>
`;
